mne/io/edf/edf.py

```diff
@@ -725,28 +730,23 @@ def _read_gdf_header(fname, stim_channel, exclude):
             etp = header_nbytes + n_records * edf_info['bytes_tot']
             # skip data to go to event table
             fid.seek(etp)
-            etmode = fid.read(1).decode()
-            if etmode != '':
-                etmode = np.fromfile(fid, np.uint8, 1).tolist()
+            etmode = np.fromfile(fid, np.uint8, 1)[0]
+            if etmode in (1, 3):
```

Comment:
I didn't find any docs for versions < 2, but these changes make sense to me and works with our testing data.